decidim-elections/app/commands/decidim/votings/census/admin/create_dataset.rb

```diff
@@ -45,11 +46,20 @@ def create_census_dataset!
                 organization: form.current_participatory_space.organization,
                 voting: form.current_participatory_space,
                 file: form.file.original_filename,
-                status: :review_data
+                csv_row_raw_count: csv_row_count,
+                status: :creating_data
               },
               visibility: "admin-only"
             )
           end
+
+          def csv_rows
```

Comment:
💬  I'm not sure about this, but maybe for very big files we need to do something like `wc -l #{form.file.tempfile.path}` to avoid loading all the lines in the server memory. I have tested it with files of almost 2GB and it takes less than 2 seconds. I don't know if this is too hacky or if there is any better way to this but it would be nice to consider it. What do you think?

Reply:
I can change it to what you suggest if it is more performant. 😁

Reply:
If you are okay we can work _this_ and the _increment counter_ out in another PR, so we don't block other PR's that depend on this one.